tests/phpunit/api/v3/MailingTest.php

```diff
@@ -288,14 +289,14 @@ public function testMailerPreview() {
     $this->assertDBQuery($maxIDs['recipient'], 'SELECT MAX(id) FROM civicrm_mailing_recipients');
     $baseurl = CRM_Utils_System::baseCMSURL();
     $previewResult = $result['values'][$result['id']]['api.Mailing.preview'];
-    $this->assertEquals("Hello $displayName", $previewResult['values']['subject']);
+    $this->assertEquals("[CiviMail Draft] Hello $displayName", $previewResult['values']['subject']);
```

Comment:
@totten @eileenmcnaughton, this one I find interesting in so far as is this a contract change with the API? it feels like maybe there was a difference in preview v test before but with flexmailer involved that goes away?

Reply:
@totten I think it's a feature that was added to flexmailer & not BAO_Mailing (that's the sort of thing that has made this duplication of functionality painful)

Note I just rebased this PR - but I haven't fixed the last 2 tests - or figured out what is required for them

Reply:
Oh, the `[CiviMail Draft]` feature definitely existed before in a general sense (eg [here](https://github.com/civicrm/civicrm-core/blob/67d870c5cf52f5a93d554d3c136596f53ee10e27/CRM/Mailing/BAO/MailingJob.php#L506-L508) vs [here](https://github.com/civicrm/civicrm-core/blob/bdf67e2837be3eec51cf647cb974c05daad453e5/ext/flexmailer/src/Listener/TestPrefix.php#L22)). I think Seamus's analysis is correct that there's a subtle change.

You can view it as a grid:

| Use Case | Action | BAO Behavior | Flexmailer Behavior |
| --| --| --| --|
| Send a test mailing over SMTP | `send_test` | Has prefix | Has prefix |
| Preview a test mailing in browser | `preview` | Omits prefix | Has prefix |

For Flexmailer, this implies that the flag `is_test` is getting set in both cases. So this makes two questions:

* Which behavior(s) do we accept
* If we want to change Flexmailer, then we probably need some change to `is_test` (either don't emit `is_test` for previews, or use some additional/alternative signal so that`TestPrefix` can distinguish the two)

Reply:
@totten so I think the BAO behaviour seems more correct to me here reading ^^ & imagining what I would expect as a user. OTOH I think it would be fine to go with flexmailer behaviour if it required more than a trivial amount of work to change

Reply:
In light local testing, this gets past the `[CiviMail Draft]` error with the original assertion.

```diff
diff --git a/ext/flexmailer/src/API/MailingPreview.php b/ext/flexmailer/src/API/MailingPreview.php
index cdd11ac428..63c90fc4ca 100644
--- a/ext/flexmailer/src/API/MailingPreview.php
+++ b/ext/flexmailer/src/API/MailingPreview.php
@@ -43,7 +43,6 @@ class MailingPreview {

     $job = new \CRM_Mailing_BAO_MailingJob();
     $job->mailing_id = $mailing->id ?: NULL;
-    $job->is_test = 1;
     $job->status = 'Complete';
     // $job->save();
```

I can't see any obvious side-effects here (eg from grepping or recollection). The thing is that the record isn't supposed to be saved. If there was some kind of bug that caused it to be saved, then `is_test=1` would probably be more accurate. But on the surface, I can't see any downside to having it mocked as `is_test=0`.

Reply:
@totten I guess if 'rollback isn't working' then there is a problem - but if we thought that was happening we'd use a different method wouldn't we?

Reply:
@eileenmcnaughton  Also, AFAIK, it's not supposed to be using rollback right now. (It'd take a bit to trace when it was used - but in the current code for CiviMail+Flexmailer+Mosaico, I don't see it being used.) The only failure case would be if some other party (extension) issued `getJob()->save()`, though it's hard to see why that would happen.

I've opened #21715 to make Flexmailer's behavior match core's behavior and to refine the non-saving

Reply:
Eileen reviewed/merged 21715. This PR is now uses the original assertions (ie omitting `[CiviMail Draft]` during `preview`). So I think we can resolve the subthread.